r-package/grf/bindings/RegressionForestBindings.cpp

```diff
@@ -48,17 +48,23 @@ Rcpp::List regression_train(Rcpp::NumericMatrix train_matrix,
                             unsigned int samples_per_cluster,
                             bool compute_oob_predictions,
                             unsigned int num_threads,
-                            unsigned int seed) {
+                            unsigned int seed,
+                            bool mbb,
+                            unsigned int blocklength,
+                            unsigned int blocknum) {
   ForestTrainer trainer = regression_trainer();
 
   std::unique_ptr<Data> data = RcppUtilities::convert_data(train_matrix, sparse_train_matrix);
   data->set_outcome_index(outcome_index - 1);
-  if(use_sample_weights) {
-      data->set_weight_index(sample_weight_index - 1);
+  
+  ForestOptions options(num_trees, ci_group_size, sample_fraction, mtry, min_node_size, honesty,
+                        honesty_fraction, honesty_prune_leaves, alpha, imbalance_penalty, num_threads, seed, clusters, samples_per_cluster);
+  if(mbb){
+      ForestOptions options(num_trees, ci_group_size, sample_fraction, mtry, min_node_size, honesty,
```

Comment:
This does not since the variable "options" is only valid inside this if (See https://en.cppreference.com/w/cpp/language/scope). It needs to be declared outside so that the following code can access that.

You need to separate declaration (outside if) and assignment (inside if) in this case.

Reply:
> This does not since the variable "options" is only valid inside this if (See https://en.cppreference.com/w/cpp/language/scope). It needs to be declared outside so that the following code can access that.
> 
> You need to separate declaration (outside if) and assignment (inside if) in this case.

I feel confused. I have declared both (outside if) and (inside if) differently. As the scope document says, when mbb=T,  the inside "options" will be invalid after the "if(mbb){...}" part. What do you mean by separate declaration?

Reply:
A variable cannot be declared twice. Here you actually declared two different variables, and only the one outside will survive.

you can do this:
if (mbb) {
 options = ForestOptions{num_trees, ...};
}
so that it is an assignment instead of a declaration. it overwrites the one declared outside.

Reply:
Thanks, I see. Let me try it.